src/CMakeLists.txt

```
@@ -226,6 +232,15 @@ if(Omega_h_USE_SimModSuite)
   target_link_libraries(omega_h PUBLIC "${SIMMODSUITE_LIBS}")
 endif()
 
+if(Omega_h_USE_ADIOS2)
+  if (Omega_h_USE_MPI)
+    target_include_directories(omega_h PRIVATE "${ADIOS2_INCLUDE_DIRS}")
+    target_link_libraries(omega_h PUBLIC "${ADIOS2_LIBRARIES}")
+    target_link_libraries(omega_h PUBLIC MPI::MPI_CXX)
+  else()
+    message("set Omega_h_USE_MPI to use ADIOS2")
```

```suggestion
    message(FATAL_ERROR, "set Omega_h_USE_MPI to use ADIOS2")
```
The cmake configure should fail if a dependency is missing:
https://cmake.org/cmake/help/latest/command/message.html#general-messages

If the current implementation of the adios2 interface doesn't support GPU execution there should also be a conditional for that.  Something like the following should work:

```
if(Omega_h_USE_Kokkos)
  foreach(backend IN LISTS "CUDA" "HIP" "SYCL")
    if (backend IN_LIST Kokkos_DEVICES)
      message(FATAL_ERROR, "Omega_h_USE_ADIOS2 does not support GPU backends")
    endif()
  endforeach()
endif()
```